File: backend/app/models/employee_attendance.py
# backend/app/models/employee_attendance.py
import uuid
from sqlalchemy import Column, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy import TIMESTAMP
from sqlalchemy.orm import relationship
from sqlalchemy import ForeignKey

from app.db.base import Base


class EmployeeAttendance(Base):
    __tablename__ = "employee_attendance"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, server_default=text("gen_random_uuid()"))
    employee_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    shop_id = Column(UUID(as_uuid=True), ForeignKey("shops.id"), nullable=False)
    checkin_at = Column(TIMESTAMP(timezone=True), server_default=func.now(), nullable=False)
    checkout_at = Column(TIMESTAMP(timezone=True), nullable=True)
    status = Column(Text, nullable=False, server_default="'present'::character varying")
    note = Column(Text, nullable=True)
    created_at = Column(TIMESTAMP(timezone=True), server_default=func.now(), nullable=False)

    employee = relationship("User", back_populates="attendance")
    shop = relationship("Shop", back_populates="employee_attendance")

File: j-backend/src/models/EmployeeAttendance.js
/**
 * EmployeeAttendance model - Employee check-in/out records.
 */
const { DataTypes } = require('sequelize');
const { sequelize } = require('../db');

const EmployeeAttendance = sequelize.define('EmployeeAttendance', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  employee_id: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: 'users',
      key: 'id',
    },
  },
  shop_id: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: 'shops',
      key: 'id',
    },
  },
  checkin_at: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW,
  },
  checkout_at: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  status: {
    type: DataTypes.TEXT,
    allowNull: false,
    defaultValue: 'present',
  },
  note: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
}, {
  tableName: 'employee_attendance',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: false,
});

module.exports = EmployeeAttendance;

